aws-android-sdk-kinesisvideo/src/test/java/com/amazonaws/mobileconnectors/kinesisvideo/mediasource/file/ImageFileMediaSourceConfigurationTest.java

```diff
@@ -0,0 +1,101 @@
+package com.amazonaws.mobileconnectors.kinesisvideo.mediasource.file;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import androidx.test.core.app.ApplicationProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(RobolectricTestRunner.class)
+public class ImageFileMediaSourceConfigurationTest {
+
+    private Context context;
+    private AssetManager assetManager;
+
+    final int FPS = 25;
+    final String FRAME_DIR = "test_folder";
+    final String FILE_NAME_FORMAT = "frame-%03d.h264";
+    final int START_FILE_INDEX = 1;
+    final int END_FILE_INDEX = 375;
+
+    @Before
+    public void initialize() {
+        context = ApplicationProvider.getApplicationContext();
+        assetManager = context.getAssets();
+    }
+
+    @Test
+    public void testConfigurationFieldsAreSetCorrectly() {
+        ImageFileMediaSourceConfiguration config = new ImageFileMediaSourceConfiguration.Builder()
+                .fps(FPS)
+                .dir(FRAME_DIR)
+                .assetManager(assetManager)
+                .filenameFormat(FILE_NAME_FORMAT)
+                .startFileIndex(START_FILE_INDEX)
+                .endFileIndex(END_FILE_INDEX)
+                .build();
+
+        assertThat(config.getFps()).isEqualTo(FPS);
+        assertThat(config.getDir()).isEqualTo(FRAME_DIR);
+        assertThat(config.getAssetManager()).isSameInstanceAs(assetManager);
+        assertThat(config.getFilenameFormat()).isEqualTo(FILE_NAME_FORMAT);
+        assertThat(config.getStartFileIndex()).isEqualTo(START_FILE_INDEX);
+        assertThat(config.getEndFileIndex()).isEqualTo(END_FILE_INDEX);
+    }
+
+    @Test
+    public void testEmptyFilePathIsValid() {
+        ImageFileMediaSourceConfiguration config = new ImageFileMediaSourceConfiguration.Builder()
+                .fps(FPS)
+                .dir("")
```

Comment:
are we considering to read the images from current directory in this setting?

Reply:
Yes. See https://github.com/awslabs/aws-sdk-android-samples/blob/dab285be113a68daa2a72f03de52c1b4b679a2a6/AmazonKinesisVideoDemoApp/src/main/java/com/amazonaws/kinesisvideo/demoapp/fragment/FramesFragment.java#L127 for more info